spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/configclient/ZookeeperConfigServerBootstrapper.java

```diff
@@ -123,11 +79,19 @@ public void initialize(BootstrapRegistry registry) {
 
 		// promote beans to context
 		registry.addCloseListener(event -> {
-			ZookeeperDiscoveryClient discoveryClient = event.getBootstrapContext().get(ZookeeperDiscoveryClient.class);
```

Comment:
Since the beans aren't registered in the bootstrap context we need to get them from `ZookeeperFunction` to promote them

Reply:
It was done out of necessity.  As I described above the `Binder` does not contain the property sources to correctly resolve the `isEnabled` call when Boot is trying to load all its configuration, so this code no longer worked.